packages/app/src/components/inspector/ColorInputFieldWithPicker.tsx

```diff
@@ -52,6 +76,72 @@ interface Props {
   onChange: (color: Sketch.Color) => void;
 }
 
+interface GridProps {
+  id?: string;
+  swatches: Sketch.Swatch[];
+  layout: SwatchLayout;
+  onClick: (color: Sketch.Color) => void;
+}
+
+const SwatchesSelector = memo(function SwatchesSelector({
+  swatches,
+  layout,
+  onClick,
+}: GridProps) {
+  const [selected, setSelected] = useState<string>('');
+
+  return (
+    <>
+      {layout === 'grid' && (
+        <GridSmall>
+          {swatches.map((item) => {
+            const colorString = hsvaToRgbaString(
+              rgbaToHsva(sketchColorToRgba(item.value)),
+            );
+
+            return (
+              <Square
+                key={item.do_objectID}
+                color={colorString}
+                selected={selected === item.do_objectID}
+                onClick={() => {
+                  onClick(item.value);
```

Comment:
Ok - I don't think I explained this part very well... colors have a property called `swatchID` which we should set when a swatch is selected. For example:

```ts
onClick({
  ...item.value,
  swatchID: item.do_objectID,
});
```

We can then use this to show selection. So for example, `GridProps` could contain a `selectedSwatchId` prop, and then 

```tsx
<Square 
  selected={selectedSwatchId === item.do_objectID} 
  // ... 
/>
```

Then in the `ColorInputFieldWithPicker` we'll render using 

```tsx
<SwatchesSelector
  selectedSwatchId={value.swatchID} 
  // ...
/>
```

Hopefully that's clear - if not let me know